framework/lib/components/combo-picker/combo-picker.tsx

```diff
@@ -87,10 +90,24 @@ export const ComboPicker = ({
   }, [ enableSelectInput ])
 
   useEffect(() => {
-    if (!is(Number, inputValue) && defaultToZeroIfEmpty) {
-      handleInputChange(clamp(min, max, 0))
+    const needsToCorrectOption = !selectOption
+    const needsToCorrectInput = !is(Number, inputValue) && defaultToZeroIfEmpty
+
+    const option = needsToCorrectOption ? options[0] : selectOption
+    const input = needsToCorrectInput ? clamp(min, max, 0) : inputValue
+
+    if (needsToCorrectOption) {
+      setselectOptionState(option)
+    }
+
+    if (needsToCorrectInput) {
+      setInputValueState(input)
+    }
+
+    if (needsToCorrectOption || needsToCorrectInput) {
+      onChange?.(getNewValue(option, input))
     }
-  }, [ valueProp?.input, defaultToZeroIfEmpty ])
+  }, [ inputValue, defaultToZeroIfEmpty ])
```

Comment:
Why listen to inputValueState here? Maybe put the clamping in the handleInputChange callback instead, so you won't have to set a state in a useEffect where that state also is a dependency

Reply:
The way it was before, it would only clamp to 0 if were recieved a new value in valueProp?.input, which is mostly the case of forms, in which when we empty the input, the value in the form will be changed and passed to the moneyInput as 0, but when the component is not inside a form or the valueProp is not controlled, if you empty the input completely it will never enter useEffect to set it to 0

Reply:
okay so you can skip this aswell